app/adapters/explorer-resource.js

```diff
@@ -77,25 +73,17 @@ var ExplorerResourceAdapter = DS.RESTAdapter.extend({
     (Most Riak cluster resources do not have globally unique IDs.
     For example, bucket types are only unique within a cluster.)
     */
-    normalizeId: function(record, type, query, idKey) {
-        var prefix = [];
-        if(query.clusterId) {
-            prefix.push(query.clusterId);
-        }
-        if(query.bucketTypeId && type.modelName !== 'bucket-type') {
-            prefix.push(query.bucketTypeId);
-        }
-        if(!idKey) {
-            idKey = 'id';
-        }
-        var originalId = record[idKey];
-        record.original_id = originalId;
-        prefix.push(originalId);
-        var compositeId = prefix.join('/');
-        record['id'] = compositeId;
-        if(record.props) {
-            record.props['id'] = compositeId;
-        }
+    normalizeId: function(record, type, query, idKey='id') {
+        let prefix = [];
+
+        if(query.clusterId) { prefix.push(query.clusterId); }
+        if(query.bucketTypeId && type.modelName !== 'bucket-type') { prefix.push(query.bucketTypeId); }
+        prefix.push(record[idKey]);
+
+        record.original_id = record[idKey];
+        record.id = prefix.join('/');
+
+        if(record.props) { record.props.id = record.id; }
```

Comment:
Just realized I need to add in comments for this. Should this be in bucket-types adapter? Seems like this logic is strictly for them